### ISSUE TITLE
[SETUP ISSUE] remove_hosts_entries throws sed: -e expression #1, char 1: unknown command: `.'

### ISSUE THREAD

**Issue by karser** [lucatume/wp-browser#286]
**Environment**
OS: Ubuntu 18.04
PHP version: unrelated
Installed Codeception unrelated
Installed wp-browser version: 2.2.18
WordPress version: unrelated
Local development environment: docker
WordPress structure and management: docker

Hello, I'm crafting the CI/CD makefile for my plugin which eventually will be running on travis. First I'd like to make it work locally.

I'm getting this error when running `make sync_hosts_entries`:
```
echo "Removing project wp-browser hosts entries..."
Removing project wp-browser hosts entries...
sudo sed -n -i .orig '/## wp-browser project - start ##/{x;d;};1h;1!{x;p;};${x;p;}' /etc/hosts
sed: -e expression #1, char 1: unknown command: `.'
Makefile:78: recipe for target 'remove_hosts_entries' failed
make: *** [remove_hosts_entries] Error 1
```

Do you think it has to do with my environment?

I managed it to work on my local by replacing
```
remove_hosts_entries:
	echo "Removing project ${PROJECT} hosts entries..."
	sudo sed -n -i .orig '/## ${PROJECT} project - start ##/{x;d;};1h;1!{x;p;};$${x;p;}' /etc/hosts
	sudo sed -i .orig '/^## ${PROJECT} project - start ##/,/## ${PROJECT} project - end ##$$/d' /etc/hosts
```
with
```
remove_hosts_entries:
	echo "Removing project ${PROJECT} hosts entries..."
	sudo sed -i '/^## ${PROJECT} project - start ##/,/## ${PROJECT} project - end ##$$/d' /etc/hosts
```

**Comment by lucatume** [comment]
Hi @karser,

the `Makefile` in wp-browser root folder is really for its development purposes and very much tailored to my own needs.
I'm surprised/amazed you would run it locally.
I work on a Mac and `sed` is one of those binaries that differ between Mac and Linux systems.
If you're willing to there's no harm in trying to make this work on Linux too.

I will take a look at this and see if I can make something that works on both.

**Comment by karser** [comment]
>the Makefile in wp-browser root folder is really for its development purposes and very much tailored to my own needs.

I used to write bash scripts for docker CI/CD, but recently discovered makefiles - it's much better fit to this purpose!
Actually I was able to re-use 70-80% of your makefile. I ended up with this:
```
remove_hosts_entries:
	# backup hosts first
	sudo cp /etc/hosts /etc/hosts.$(date +%F_%R)
	echo "Removing project ${PROJECT} hosts entries..."
	sudo sed -i '/^## ${PROJECT} project - start ##/,/## ${PROJECT} project - end ##$$/d' /etc/hosts
```

>I work on a Mac and sed is one of those binaries that differ between Mac and Linux systems.

How come it works on Travis then?

**Comment by lucatume** [comment]
> I used to write bash scripts for docker CI/CD, but recently discovered makefiles - it's much better fit to this purpose!
> Actually I was able to re-use 70-80% of your makefile. I ended up with this:
> 
> ```
> remove_hosts_entries:
> 	# backup hosts first
> 	sudo cp /etc/hosts /etc/hosts.$(date +%F_%R)
> 	echo "Removing project ${PROJECT} hosts entries..."
> 	sudo sed -i '/^## ${PROJECT} project - start ##/,/## ${PROJECT} project - end ##$$/d' /etc/hosts
> ```

Great!

> How come it works on Travis then?

It doesn't; I use it only locally in the `ci_local_prepare` task, that is never called by the `.travis` configuration file.

See https://github.com/lucatume/wp-browser/blob/master/Makefile#L194

**Comment by karser** [comment]
I see, that makes sense! I'm using `ci_local_prepare` locally btw, but didn't get to travis config yet.

**Comment by lucatume** [comment]
In [this PR](https://github.com/lucatume/wp-browser/pull/288) I've simplified the script and tested it locally using `gnu-sed`.
I've kept the `-i.orig` flag since all versions of `sed` (that I'm aware of) will support replacement in place with the additional specification of a backup file extension.
Since this comes built-in with `sed` I'd keep this simpler and skip the line where the file is explicitly backed up.
